app/services/post_check_availability_task_service.rb

```diff
@@ -0,0 +1,48 @@
+class PostCheckAvailabilityTaskService < TaskService
+  def initialize(options)
+    super
+    @source = Source.find(options[:source_id]) || raise("source_id must be specified in options")
+  end
+
+  def process
+    update_source
+    create_kafka_event
+    create_refresh_upload_task
+    self
+  end
+
+  private
+
+  def update_source
+    source.update_attribute!(:input, @options["output"])
+  end
+
+  def create_kafka_event
+    if ENV['NO_KAFKA'].blank?
+      CatalogInventory::Api::Messaging.client.publish_topic(
+        :service => "platform.catalog-inventory.task-output-stream",
```

Comment:
@hsong-rh Sources is listening for this topic
https://github.com/RedHatInsights/sources-api/blob/bfdace28416d87be111af9bec064a2ef5795ab23/lib/availability_status_listener.rb#L4

Reply:
@hsong-rh 
The name of the topic is **platform.sources.status**

Data :
"resource_type":"Source",
"resource_id":"18909",
"status":"unavailable"

**Headers** : 
    
'event_type', 'availability_status'

Reply:
This is where the message is generated
https://github.com/RedHatInsights/topological_inventory-providers-common/blob/32b9b09a9a40bbaa9ae2abd4ee4177fc28ee4ed1/lib/topological_inventory/providers/common/operations/source.rb#L151

Reply:
https://github.com/RedHatInsights/topological_inventory-providers-common/blob/32b9b09a9a40bbaa9ae2abd4ee4177fc28ee4ed1/lib/topological_inventory/providers/common/operations/source.rb#L18